pkg/jsonnet/imports_test.go

```diff
@@ -23,3 +27,79 @@ func TestTransitiveImports(t *testing.T) {
 		"trees/peach.jsonnet",
 	}, imports)
 }
+
+const testFile = `
+local localImport = <IMPORT>;
+local myFunc = function() <IMPORT>;
+
+{
+	local this = self,
+
+	attribute: {
+		name: 'test',
+		value: self.name,
+		otherValue: 'other ' + self.value,
+	},
+	nested: {
+		nested: {
+			nested: {
+				nested: {
+					nested1: {
+						nested: {
+							nested1: {
+								nested: {
+									attribute: <IMPORT>,
+								},
+							},
+							nested2: {
+								strValue: this.nested.nested.nested,
+							},
+						},
+					},
+					nested2: {
+						intValue: 1,
+						importValue: <IMPORT>,
+					},
+				},
+			},
+		},
+	},
+
+	other: myFunc(),
+	useLocal: localImport,
+}`
+
+func BenchmarkGetSnippetHash(b *testing.B) {
+	// Create a very large and complex project
+	tempDir := b.TempDir()
+
+	var mainContentSplit []string
+	for i := 0; i < 1000; i++ {
+		mainContentSplit = append(mainContentSplit, fmt.Sprintf("(import 'file%d.libsonnet')", i))
+	}
+	require.NoError(b, os.WriteFile(filepath.Join(tempDir, "main.jsonnet"), []byte(strings.Join(mainContentSplit, " + ")), 0644))
+	for i := 0; i < 1000; i++ {
+		err := os.WriteFile(
+			filepath.Join(tempDir, fmt.Sprintf("file%d.libsonnet", i)),
+			[]byte(strings.ReplaceAll(testFile, "<IMPORT>", fmt.Sprintf("import 'file%d.libsonnet'", i+1))),
+			0644,
+		)
+		require.NoError(b, err)
+	}
+	require.NoError(b, os.WriteFile(filepath.Join(tempDir, "file1000.libsonnet"), []byte(`"a string"`), 0644))
```

Comment:
This will replace the last file created in the loop above with this new content, is that the idea here?